### PR TITLE
[MM-51905] Fix: Successful trial request shows error modal

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
A late small refactor broke this - the requestTrialLicense function returns 

{data: <data>} on success
or 
{error: <message>, data: { <data>}} on error

I've destructured the return value so the if block should only be entered in the event of an error rather than always.

#### Release Note

```release-note
None
```

### CODE REVIEW

File: webapp/channels/src/components/start_trial_form_modal/index.tsx [hunk 1]
```diff
@@ -129,15 +129,15 @@ function StartTrialFormModal(props: Props): JSX.Element | null {
...
+        const {error, data} = await dispatch(requestTrialLicense(trialRequestBody, props.page || 'license'));
         if (error) {
             setLoadStatus(TrialLoadStatus.Failed);
             let title;
             let subtitle;
             let buttonText;
             let onTryAgain = handleErrorModalTryAgain;
 
-            if (error?.data.status === 422) {
+            if (data.status === 422) {
```
Comment (marianunez) [inline webapp/channels/src/components/start_trial_form_modal/index.tsx:140]: No chance of `data` to be null?

**Review by marianunez — APPROVED** [review]
Thanks @nickmisasi!

File: server/channels/utils/license-public-key-test.txt [hunk 2]
```diff
@@ -1,9 +1,9 @@
 -----BEGIN PUBLIC KEY-----
-MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAyZmShlU8Z8HdG0IWSZ8r
-tSyzyxrXkJjsFUf0Ke7bm/TLtIggRdqOcUF3XEWqQk5RGD5vuq7Rlg1zZqMEBk8N
-EZeRhkxyaZW8pLjxwuBUOnXfJew31+gsTNdKZzRjrvPumKr3EtkleuoxNdoatu4E
-HrKmR/4Yi71EqAvkhk7ZjQFuF0osSWJMEEGGCSUYQnTEqUzcZSh1BhVpkIkeu8Kk
-1wCtptODixvEujgqVe+SrE3UlZjBmPjC/CL+3cYmufpSNgcEJm2mwsdaXp2OPpfn
-a0v85XL6i9ote2P+fLZ3wX9EoioHzgdgB7arOxY50QRJO7OyCqpKFKv6lRWTXuSt
```
Comment (marianunez) [inline server/channels/utils/license-public-key-test.txt:7]: Reminder to revert this before merging

**Review by neallred — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
Hi @nickmisasi ,
I've tested the request trial and upgrade flow.
- request trial was successful 
Screenshot 2023-04-04 at 10 43 18 AM

- the purchase was throwing error, would it be due to few test changes ?
Screenshot 2023-04-04 at 10 47 04 AM

**Comment by nickmisasi** [comment]
@yasserfaraazkhan some overrides are required to get self hosted purchase to work in spinwick's like this, so the error you got is expected. The fact you got the trial benefits modal after activating the trial means we're good to go here!

**Review by yasserfaraazkhan — APPROVED** [review]
thank you Nick 🎉

**Pull request merged**